### DISCUSSION TITLE
issue with snapshot on Esp32cam with tasmota32 webcam firmware

### DISCUSSION THREAD

**Discussion by HerrTuulik** [arendst/Tasmota#17290] [Q&A]
Hello!

I've flashed "Tasmota32 Webcam" to esp32cam.
I've got a node-red setup where i call a http://deviceIP:PORT/snapshot.jpg and parse it to telegram via bot.
When i have the stream open somewhere and i call the snapshot, it gives me what i expect: the latest image.
When i don't have the stream open and i call it, it gives me an old image. From there, every consecutive call gives me a slightly refreshed image in time but not what is happening "now". So i have to spam it a bunch of times for it to refresh to the point i want.

I'm hoping somebody has some insight on how to get the latest snapshot without having to spam it or have the stream open if possible.

**Comment by philippe1480** [comment]
Hello,

I have the same problem.  Did you find a solution ?

**Reply by HerrTuulik** [reply]
Unfortunately not.

**Reply by gemu2015** [reply]
i worked on this in December. my goal was to also be able to request a selected image size.
however i delayed this because the coders of esp32cam are working on an update that should solve these problems.  so waiting for an update from their side.

**Comment by Jason2866** [comment]
@gemu2015 please keep me updated, when the IDF Cam driver has changes we want to have.
Needs updating in our fork and a new compile of the Arduino libs for the core. -> Core Update!

**Reply by gemu2015** [reply]
@Jason2866 
as there is obviously no progress from me_nodev at GitHub i introduced a workaround.

**Comment by dhavaldesai-HA** [comment]
Exactly same problem as mentioned by "HerrTuulik". I use the same setup, i.e. node-red and Telegram bot and it always gives old image and not a 'live' one. Using this setup since August 2022 and facing this issue. Is there any update?

**Reply by Jason2866** [reply]
There is a workaround in Tasmota. Use latest release or development version.

**Reply by sfromis** [reply]
If you upgrade, this fix looks relevant: https://github.com/arendst/Tasmota/pull/17840
